src/index.js

```diff
@@ -70,6 +67,64 @@ const spanning = {
   [SPANNING_MF_VAL_NONE]: ""
 };
 
+async function replace(src, modifier) {
+  const cssTextFromStyleSheet = sheet => {
+    return Array.from(sheet.cssRules).reduce((prev, cssRule) => {
+      return prev + cssRule.cssText;
+    }, "");
+  }
+
+  if ("adoptedStyleSheets" in src.shadowRoot) {
+    for (let sheet of src.shadowRoot.adoptedStyleSheets) {
+      const text = cssTextFromStyleSheet(sheet);
+      const newText = modifier(text);
+      // As these are shared across multiple elements, only
+      // update once.
+      if (text.localeCompare(newText) !== 0) {
+        sheet.replace(newText);
+      }
+    }
+  }
+
+  // As it is hard of the HTML, it needs to be rendered first.
+  await customElements.whenDefined(src.nodeName.toLowerCase());
+  for (let sheet of src.shadowRoot.querySelectorAll("style")) {
+    const text = sheet.innerText;
+    const newText = modifier(text);
+    sheet.innerText = newText;
+  }
+}
+
+export async function upgrade(element) {
+  const adjustCSS = sheet => {
+    const noSpanningCSS = replaceSpanningMediaBlocks(sheet, "");
+    const spanningCSS = getSpanningCSSText(sheet);
+
+    const name = element.nodeName.toLowerCase();
+    spanning[name] = {
+      [SPANNING_MF_VAL_HOR]: "",
+      [SPANNING_MF_VAL_VER]: "",
+      [SPANNING_MF_VAL_NONE]: ""
+    };
+
+    Object.keys(spanningCSS).forEach(k => {
+      if (typeof spanning[name][k] !== typeof(undefined)) {
+        spanning[name][k] += `
+          /* origin: ${element.nodeName.toLowerCase()} */
+          ${spanningCSS[k]}`;
+      }
+    });
+  
+    return noSpanningCSS;
+  }
+
+  await replace(element, sheet => adjustCSS(sheet));
+
+  insertSpanningStyles(element);
+  window.addEventListener("resize", element => debounce(insertSpanningStyles, 150));
```

Comment:
This won't work. It should be `window.addEventListener("resize", () => debounce(insertSpanningStyles(element), 150));`

Reply:
fixed